src/modules/copied/actions.js

```diff
@@ -1,43 +1,74 @@
-    const metric = Object.assign({}, state.metrics.find(m => m.space === spaceId && isAtLocation(m, location)))
-    const guesstimate = state.guesstimates.find(g => g.metric === metric.id)
+    const region = state.selectedRegion
+    const metrics = state.metrics.filter(m => m.space === spaceId && isWithinRegion(m.location, region))
+    const guesstimates = metrics.map(metric => state.guesstimates.find(g => g.metric === metric.id))
 
-    dispatch({type: "COPY", copied: {metric, guesstimate}})
+    dispatch({type: "COPY", copied: {metrics, guesstimates, region}})
   }
 }
 
+function translateReadableIds(input, idMap) {
+  if (!input) {return ""}
+  const re = RegExp(Object.keys(idMap).join("|"), "g")
+  return input.replace(re, (match) => idMap[match])
+}
+
 export function paste(spaceId){
   return (dispatch, getState) => {
     const state = getState()
-    if (!(state.copied && state.selection)) { return }
+    if (!(state.copied && state.selectedCell && isLocation(state.selectedCell))) { return }
+
+    const {metrics, guesstimates, region} = state.copied
+    const location = state.selectedCell
+
+    const translateFn = translate(region[0], location)
+
+    const pasteRegion = [location, translateFn(region[1])]
 
-    const {metric, guesstimate} = state.copied
+    const spaceMetrics = state.metrics.filter(m => m.space === spaceId)
+    let existingReadableIds = spaceMetrics.map(m => m.readableId)
 
-    const location = state.selection
-    const spaceMetrics = getState().metrics.filter(m => m.space === spaceId)
-    const existingReadableIds = spaceMetrics.map(m => m.readableId)
+    let readableIdsMap = {}
+    const newMetrics = _.map(metrics, metric => {
+      const newMetric = Object.assign(
+        {},
+        metric,
+        e.metric.create(existingReadableIds),
+        {space: spaceId},
+        {location: translateFn(metric.location)}
+      )
+      existingReadableIds.push(newMetric.readableId)
+      readableIdsMap[metric.readableId] = newMetric.readableId
+      return newMetric
+    })
 
-    const newItem = Object.assign({}, metric, e.metric.create(existingReadableIds), {location})
+    const newGuesstimates = _.map(
+      guesstimates,
+      (guesstimate, i) => Object.assign(
+        {},
+        guesstimate,
+        {metric: newMetrics[i].id},
+        {input: translateReadableIds(guesstimate.input, readableIdsMap)}
+      )
+    )
 
-    const newGuesstimate = Object.assign({}, guesstimate, {metric: newItem.id})
+    const existingMetrics = spaceMetrics.filter(m => isWithinRegion(m.location, pasteRegion))
+    existingMetrics.forEach(existingMetric => {dispatch(metricActions.removeMetric(existingMetric.id))})
 
-    const existingMetric = spaceMetrics.find(m => isAtLocation(m, location))
-    if (existingMetric) {
-      dispatch(metricActions.removeMetric(existingMetric.id))
-    }
+    newMetrics.forEach((newMetric, i) => {dispatch({type: 'ADD_METRIC', item: newMetric, newGuesstimate: newGuesstimates[i]})})
```

Comment:
Agreed, going to leave off for now though as I don't think there is significant functionality we can extract here that wouldn't be just extracting function scope and putting it higher in the file, which IMO would increase total cost of ownership.